homeassistant/components/knx/__init__.py

```diff
@@ -407,10 +447,8 @@ def register_event_callback(self) -> TelegramQueue.Callback:
         """Register callback for knx_event within XKNX TelegramQueue."""
         # backwards compatibility for deprecated CONF_KNX_EVENT_FILTER
         # use `address_filters = []` when this is not needed anymore
-        address_filters = list(
-            map(AddressFilter, self.config[DOMAIN][CONF_KNX_EVENT_FILTER])
-        )
-        for filter_set in self.config[DOMAIN][CONF_EVENT]:
+        address_filters = list(map(AddressFilter, self.config[CONF_KNX_EVENT_FILTER]))
+        for filter_set in self.config[CONF_EVENT]:
             _filters = list(map(AddressFilter, filter_set[KNX_ADDRESS]))
             address_filters.extend(_filters)
```

Comment:
You can pass generators directly to methods that consume iterables 
```suggestion
            address_filters.extend(map(AddressFilter, filter_set[KNX_ADDRESS]))
```

Reply:
Or like this
```suggestion
            address_filters.extend(AddressFilter(val) for val in filter_set[KNX_ADDRESS])
```

Reply:
You keep marking comments on the usage of functional approach as resolved without actually commenting or addressing

Reply:
Oh sorry! `_filters` is still used in the rest of the code. Changing this would break it.

https://github.com/home-assistant/core/blob/7a06cd86ca0a88589f0318408c7de33f10218c0b/homeassistant/components/knx/__init__.py#L458